jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/redshift/RedshiftSqlDialect.java

```diff
@@ -138,4 +120,11 @@ public String getStringLiteral(final String value) {
     public SqlGenerationVisitor getSqlGenerationVisitor(final SqlGenerationContext context) {
         return new RedshiftSqlGenerationVisitor(this, context);
     }
-}
+
+    @Override
+    protected RemoteMetadataReader createRemoteMetadataReader() {
+        assert (this.connection != null);
```

Comment:
Change `assert`-s to exceptions.

Reply:
Removed asserts since they make unit tests more complicated than necessary.